src/sentry/runner/commands/upgrade.py

```diff
@@ -1,29 +1,134 @@
 from __future__ import absolute_import, print_function
 
 import click
-import django
+import six
 
+from django.db import connections, ProgrammingError
 from django.conf import settings
 from sentry.runner.decorators import configuration
 
-DJANGO_17 = django.VERSION[0] > 1 or (django.VERSION[0] == 1 and django.VERSION[1] >= 7)
+# List of migrations which we'll fake if we're coming from South
+DJANGO_MIGRATIONS = (
+    ("contenttypes", "0001_initial"),
+    # Note that we need to have ops make the `django_content_type.name` column nullable.
+    # This migration removes the column in the same migration, which will likely take
```

Comment:
Would this be a concern for on-premise too?

Reply:
We typically don't care about this kind of thing for on-prem, as far as I know. They likely don't have a 20-30 minute deploy cycle, and so the window for errors is much shorter.